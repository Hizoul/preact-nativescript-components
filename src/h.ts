import * as preact from "preact"
import { Preact } from "preact-to-nativescript"

const customH: any = Preact.h

export default customH
