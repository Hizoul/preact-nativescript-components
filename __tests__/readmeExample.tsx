import * as renderToJson from "preact-render-to-json"
import {
  ActionBar, ActionItem, Button, Label, Page, ScrollView, StackLayout
} from "../src"
import h from "../src/h"

test("README Example Render", () => {
  expect(
    renderToJson(
      <Page>
        <ActionBar>
          <StackLayout>
            <Label>Custom Header</Label>
          </StackLayout>
          <ActionItem text="act" />
        </ActionBar>
        <StackLayout>
          <Label>PageContent</Label>
          <Button text="Button" />
        </StackLayout>
      </Page>
    )
  ).toMatchSnapshot("readme render")
})
