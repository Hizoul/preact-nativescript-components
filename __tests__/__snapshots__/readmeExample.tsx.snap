// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readme render 1`] = `
<page>
  <actionBar>
    <stackLayout>
      <label>
        Custom Header
      </label>
    </stackLayout>
    <actionItem
      text="act"
    />
  </actionBar>
  <stackLayout>
    <label>
      PageContent
    </label>
    <button
      text="Button"
    />
  </stackLayout>
</page>
`;
